Mathlib/GroupTheory/Perm/Support.lean

```diff
@@ -0,0 +1,670 @@
+#align equiv.perm.support_one Equiv.Perm.support_one
+
+@[simp]
+theorem support_refl : Support (Equiv.refl α) = ∅ :=
+  support_one
+#align equiv.perm.support_refl Equiv.Perm.support_refl
+
+theorem support_congr (h : f.Support ⊆ g.Support) (h' : ∀ x ∈ g.Support, f x = g x) : f = g := by
+  ext x
+  by_cases hx : x ∈ g.Support
+  · exact h' x hx
+  · rw [not_mem_support.mp hx, ← not_mem_support]
+    exact fun H => hx (h H)
+#align equiv.perm.support_congr Equiv.Perm.support_congr
+
+theorem support_mul_le (f g : Perm α) : (f * g).Support ≤ f.Support ⊔ g.Support := fun x => by
+  simp only [sup_eq_union]
+  rw [mem_union, mem_support, mem_support, mem_support, mul_apply, ← not_and_or, not_imp_not]
+  rintro ⟨hf, hg⟩
+  rw [hg, hf]
+#align equiv.perm.support_mul_le Equiv.Perm.support_mul_le
+
+theorem exists_mem_support_of_mem_support_prod {l : List (Perm α)} {x : α}
+    (hx : x ∈ l.prod.Support) : ∃ f : Perm α, f ∈ l ∧ x ∈ f.Support := by
+  contrapose! hx
+  simp_rw [mem_support, not_not] at hx⊢
+  induction' l with f l ih
+  · rfl
+  · rw [List.prod_cons, mul_apply, ih, hx]
+    simp only [List.find?, List.mem_cons, true_or]
+    intros f' hf'
+    refine' hx f' _
+    simp only [List.find?, List.mem_cons]
+    exact Or.inr hf'
+#align
+  equiv.perm.exists_mem_support_of_mem_support_prod
+  Equiv.Perm.exists_mem_support_of_mem_support_prod
+
+theorem support_pow_le (σ : Perm α) (n : ℕ) : (σ ^ n).Support ≤ σ.Support := fun _ h1 =>
+  mem_support.mpr fun h2 => mem_support.mp h1 (pow_apply_eq_self_of_apply_eq_self h2 n)
+#align equiv.perm.support_pow_le Equiv.Perm.support_pow_le
+
+@[simp]
+theorem support_inv (σ : Perm α) : Support σ⁻¹ = σ.Support := by
+  simp_rw [Finset.ext_iff, mem_support, not_iff_not, inv_eq_iff_eq.trans eq_comm, iff_self_iff,
+    imp_true_iff]
+#align equiv.perm.support_inv Equiv.Perm.support_inv
+
+-- Porting note: lower priority to avoid linter complaints about simp-normal form
+@[simp 1100]
+theorem apply_mem_support {x : α} : f x ∈ f.Support ↔ x ∈ f.Support := by
+  rw [mem_support, mem_support, Ne.def, Ne.def, not_iff_not, apply_eq_iff_eq]
+#align equiv.perm.apply_mem_support Equiv.Perm.apply_mem_support
+
+-- Porting note: lower priority to avoid linter complaints about simp-normal form
+@[simp 1200]
+theorem pow_apply_mem_support {n : ℕ} {x : α} : (f ^ n) x ∈ f.Support ↔ x ∈ f.Support := by
+  induction' n with n ih
+  · rfl
+  rw [pow_succ, Perm.mul_apply, apply_mem_support, ih]
+#align equiv.perm.pow_apply_mem_support Equiv.Perm.pow_apply_mem_support
+
+-- Porting note: lower priority to avoid linter complaints about simp-normal form
+@[simp 1200]
+theorem zpow_apply_mem_support {n : ℤ} {x : α} : (f ^ n) x ∈ f.Support ↔ x ∈ f.Support := by
+  cases n
+  · rw [Int.ofNat_eq_coe, zpow_ofNat, pow_apply_mem_support]
+  · rw [zpow_negSucc, ← support_inv, ← inv_pow, pow_apply_mem_support]
+#align equiv.perm.zpow_apply_mem_support Equiv.Perm.zpow_apply_mem_support
```

Comment:
Here again, I didn't find a way to make one of them obsolete by explicitly writing the correct coercion.